### ISSUE TITLE
Make the runToFix message customizable in SpotlessCheck

### ISSUE THREAD

**Issue by dweiss** [diffplug/spotless#1175]
There's been some broad discussion in the Lucene project about the sense of having per-project messages in SpotlessCheck:
```
				throw new GradleException(DiffMessageFormatter.builder()
						.runToFix("Run '" + calculateGradleCommand() + " " + getTaskPathPrefix() + "spotlessApply' to fix these violations.") 
```
Many people expressed the opinion that this is just confusing for new users (you fix one violation in one module, then you get an exception elsewhere) and the message should just be "global". I wonder if it'd be possible to add a hook that could customize the returned message (perhaps a hook accepting DiffMessageFormatter.Builder)? Then you could set runToFix to a project-specific message (or task).

This is a suggestion but it is really difficult to work around in any other way. If you think it'd be acceptable, I can provide a patch. Thank you.

**Comment by nedtwigg** [comment]
Thanks for reporting! Reading the linked PR, I'm already convinced that we should suggest `spotlessApply` instead of `:project:spotlessApply`. Would that alone solve your problem? We *could* add a hook to allow users to change it to e.g. `tidy`, but I'd like our API to be as small as possible and that seems like a fairly niche use-case which you've already been able to work around.

**Comment by dweiss** [comment]
I think there are benefits to customization of that message other than just the task name - you could point at a project-specific help resource, which explains how the project manages code formatting, etc. So I still tend to think it'd be beneficial to be able to somehow tune it on a per-project basis. a top-level spotlessApply is fine too, of course, but I can already think of another project I'm involved in where the "tidy" task invokes not just spotless but also other, non-spotless cleanups... 

Just FYI: I've looked at various ways to try to intercept those exceptions and just change the message but this isn't at all trivial (more like impossible, especially with newer gradle releases).

**Comment by nedtwigg** [comment]
You've convinced me, I'd be happy to merge a PR to allow this sort of customization.

**Comment by dweiss** [comment]
Thank you! I'll work on this in a spare moment and will provide a pr for review.

**Comment by dweiss** [comment]
Hmm... I get a consistent failure on clean main (beefy Windows machine) with:
```
gradlew -p plugin-gradle test --tests ConfigurationCacheTest.jvmLocalCache
```
the forked gradle process succeeds while it should be failing, it seems:
```
> Task :plugin-gradle:test

com.diffplug.gradle.spotless.ConfigurationCacheTest

  Test jvmLocalCache() FAILED (7.4s)

  org.gradle.testkit.runner.UnexpectedBuildSuccess: Unexpected build execution success in [snip] with arguments [spotlessApply, --stacktrace]

  Output:
  > Task :spotlessInternalRegisterDependencies UP-TO-DATE
  > Task :spotlessJava
  > Task :spotlessJavaApply
  > Task :spotlessApply

  BUILD SUCCESSFUL in 2s
  3 actionable tasks: 2 executed, 1 up-to-date
      at app//com.diffplug.gradle.spotless.ConfigurationCacheTest.jvmLocalCache(ConfigurationCacheTest.java:93)
```

Ignoring for now.

**Comment by dweiss** [comment]
I created a PR. Interestingly, when I run the top-level ./gradlew check, I get loads of formatting errors on unrelated files:
```
          -·*·Copyright·2016-2021·DiffPlug
          +·*·Copyright·2016-2022·DiffPlug
```
I suspect something something is wrong with the build cache configuration - it seems unrelated. I didn't run spotlessApply because this corrects files all across the board and it's probably not something you'd want?

**Comment by dweiss** [comment]
Thank you, @nedtwigg ! Is there anything needed in particular for the check to pass (without build caches, etc)? It's a bit strange that a clean checkout doesn't pass all the tests/ validation.

**Comment by nedtwigg** [comment]
I can't replicate the issue with erroneous copyright headers. My hunch is that it's a problem with JGit and newer working copy formats, but I don't know. Worth its own issue if it becomes a problem for you.

**Comment by dweiss** [comment]
I think I know what caused this - I used my own fork to create a PR. When you clone your own fork, origin/main may be out of sync (before the feature branch you're contributing). This causes ratchetFrom to go haywire. A clean checkout indeed passes tests but this won't:
```
git clone git@github.com:dweiss/spotless.git -b customRunToFix
cd spotless
./gradlew spotlessCheck
```

That configuration cache test still fails for me (on beefy windows machine) - can't say whether it's a race condition somewhere or if it's the environment itself. On Linux it seems to work fine.

**Comment by nedtwigg** [comment]
Published in `plugin-gradle 6.5.0`.
